test-runner-frontend/src/Views/DashboardUtils.js

```diff
@@ -0,0 +1,185 @@
+const HIGHER_IS_BETTER = 'higherIsBetter';
+const LOWER_IS_BETTER = 'lowerIsBetter';
+
+const modeData = {
+  thrpt: HIGHER_IS_BETTER,
+  avgt: LOWER_IS_BETTER,
+  sample: LOWER_IS_BETTER,
+  ss: LOWER_IS_BETTER,
+  all: LOWER_IS_BETTER,
+};
+
+export const getMachineNames = (testRunResults) => {
+  var machines = [];
+  testRunResults?.forEach((testRunResult) => {
+    Object.keys(testRunResult?.status).forEach((machine) => {
+      if (!machines.includes(machine)) {
+        machines.push(machine);
+      }
+    });
+  });
+
+  return machines;
+};
```

Comment:
this function looks like its trying to work out a unique list of machine names. it might be worth looking into using a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and then at the very end using `Array.from(yourSet)` to get an array version of the names

Reply:
Done.